tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -730,50 +600,111 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_stuffer_free(&wire_identities_in));
         }
 
-        /* Receive a list with a match later in the list */
+        /* Customer selection logic: customer rejects all identities */
         {
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_config_set_psk_selection_callback(config, s2n_test_error_select_psk_identity_callback));
+
             struct s2n_connection *conn;
             EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
 
             struct s2n_stuffer wire_identities_in = { 0 };
-            EXPECT_SUCCESS(s2n_stuffer_alloc(&wire_identities_in, sizeof(wire_identities)));
-            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&wire_identities_in, wire_identities, sizeof(wire_identities)));
+            EXPECT_ERROR_WITH_ERRNO(s2n_client_psk_recv_identity_list(conn, &wire_identities_in),
+                    S2N_ERR_UNIMPLEMENTED);
+            EXPECT_EQUAL(conn->psk_params.chosen_psk, NULL);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_config_free(config));
+            EXPECT_SUCCESS(s2n_stuffer_free(&wire_identities_in));
+        }
+
+        /* Customer selection logic: customer selects valid identity */
+        {
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_config_set_psk_selection_callback(config, s2n_test_select_psk_identity_callback));
+
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
 
             struct s2n_psk *match_psk = NULL;
             EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void**) &match_psk));
             EXPECT_OK(s2n_psk_init(match_psk, S2N_PSK_TYPE_EXTERNAL));
             EXPECT_SUCCESS(s2n_psk_set_identity(match_psk, test_bytes_data, sizeof(test_bytes_data)));
 
+            struct s2n_stuffer wire_identities_in = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&wire_identities_in, 0));
+            EXPECT_OK(s2n_write_test_identity(&wire_identities_in, &match_psk->identity));
+
+            s2n_test_customer_wire_index_choice = 0;
             EXPECT_OK(s2n_client_psk_recv_identity_list(conn, &wire_identities_in));
-            EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, wire_identities_match_index);
             EXPECT_EQUAL(conn->psk_params.chosen_psk, match_psk);
+            EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, 0);
```

Comment:
`EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, s2n_test_customer_wire_index_choice);`

Reply:
I prefer just 0. s2n_test_customer_wire_index_choice is intended to communicate the choice to the callback, and theoretically the callback could reset it or something.